### PR TITLE
Add overloads with `Duration` to `Cached` helper class

### PR DESCRIPTION
**Pull request by avdv**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you [squashed your commits]? (Optional, but makes merge messages nicer.)
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [ ] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Purpose

This change simply adds methods to the `Cached` helper class to use `Duration` as argument.

## Background Context

It seems a bit odd that one has to specify an Integer for the duration of the `Cached` helper when the underlying `CacheBuilder` operates with `Duration`.

Specifically, we want to configure the duration from the config:

```
cache.max = 5 minutes
```

```scala
val duration = config.get[Duration]("cache.max")

cached(_ => "key", duration.toSeconds.toInt) // yikes
```

### CODE REVIEW

**Comment by marcospereira** [comment]
Build is broken for Scala 2.11:

https://travis-ci.org/playframework/playframework/jobs/260201451#L1185

**Comment by avdv** [comment]
I'll have a look...

File: framework/src/play-cache/src/main/scala/play/api/cache/Cached.scala [hunk 1]
```diff
@@ -82,12 +92,24 @@ class Cached @Inject() (cache: AsyncCacheApi)(implicit materializer: Materialize
...
+    empty(key).default(duration)
+
+  /**
    * Caches the specified status, for the specified number of seconds
    */
   def status(key: RequestHeader => String, status: Int, duration: Int): CachedBuilder =
     empty(key).includeStatus(status, Duration(duration, SECONDS))
 
   /**
+   * Caches the specified status, for the specified number of seconds
```
Comment (mkurz) [inline framework/src/play-cache/src/main/scala/play/api/cache/Cached.scala]: Comment should be `..., for the specified duration`

**Comment by mkurz** [comment]
@avdv What is the status here? Do you think you can finish your contribution? Thanks!

**Comment by avdv** [comment]
> What is the status here? Do you think you can finish your contribution? Thanks!

Sorry, I am pretty loaded. I will try to finish it in the next three days.

**Comment by avdv** [comment]
@mkurz I have updated the PR according to the comments, please have a look again.

**Comment by mkurz** [comment]
Should we backport?

**Comment by marcospereira** [comment]
@mkurz see #7765.

**Pull request merged**